ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -212,19 +212,23 @@ public void runNonInteractively(final String input) throws Exception {
     }
   }
 
+  @SuppressWarnings("RedundantStringOperation") // Incorrect warning. Operation is not redundant
   private List<String> getLogicalLines(final String input) {
     // TODO: Convert the input string into an InputStream, then feed it to the terminal via
     // TerminalBuilder.streams(InputStream, OutputStream)
     final List<String> result = new ArrayList<>();
     StringBuilder logicalLine = new StringBuilder();
     for (final String physicalLine : input.split("\n")) {
-      if (!physicalLine.trim().isEmpty()) {
-        if (physicalLine.endsWith("\\")) {
-          logicalLine.append(physicalLine.substring(0, physicalLine.length() - 1));
-        } else {
-          result.add(logicalLine.append(physicalLine).toString().trim());
-          logicalLine = new StringBuilder();
-        }
+      final String trimmed = physicalLine.trim();
+      if (trimmed.isEmpty()) {
```

Comment:
Can we remove this `if` statement if we change the else below to be `else if (!trimmed.isEmpty())` ?

I find continue offensive! ;-)